src/client.rs

```diff
@@ -64,12 +71,19 @@ impl Client {
                 client_response::Which::Proposal(Ok(response)) => {
                     match response.which().unwrap() {
                         proposal_response::Which::Success(()) => {
+                            scoped_debug!("recieved response Success");
                             self.leader_connection = Some(connection);
                             return Ok(())
                         },
-                        proposal_response::Which::UnknownLeader(()) => (),
+                        proposal_response::Which::UnknownLeader(()) => {
+                            scoped_debug!("recieved response UnknownLeader");
+                            ()
+                        },
                         proposal_response::Which::NotLeader(leader) => {
-                            let connection: TcpStream = try!(TcpStream::connect(try!(leader)));
+                            scoped_debug!("recieved response NotLeader");
```

Comment:
Technically this could loop forever, since two servers could each thing the other is the leader, and ping pong the client back and forth.   Maybe not worth worrying about though, I can't think of a simple solution.